task/jefe/src/main.rs

```diff
@@ -232,6 +232,48 @@ impl idl::InOrderJefeImpl for ServerImpl<'_> {
                 self.dump_areas = dump::initialize_dump_areas();
                 Ok(())
             }
+
+            fn dump_task(
+                &mut self,
+                _msg: &userlib::RecvMessage,
+                task_index: u32,
+            ) -> Result<u8, RequestError<DumpAgentError>> {
+                // `dump::dump_task` doesn't check the task index, because it's
+                // normally called by a trusted source; we'll do it ourself.
+                if task_index == 0 {
+                    // Can't dump the supervisor
+                    return Err(DumpAgentError::NotSupported.into());
+                } else if task_index as usize > self.task_states.len() {
```

Comment:
It seems like all of these comparisons should be `>=`

Reply:
Yes, fixed in bc20ab50e8f282549365ab9ddd3c132100513f36